geometric_kernels/sampling/probability_densities.py

```diff
@@ -227,3 +234,40 @@ def base_sampler(key):
             samples.append(sign * proposal)
     samples = B.reshape(B.concat(*samples), *size)
     return key, B.cast(B.dtype(L), samples)
+
+
+def spd_density_sample(key, size, params, degree, rho):
+    nu = params["nu"]
+    L = params["lengthscale"]
+
+    samples = []
+    while len(samples) < reduce(operator.mul, size, 1):
+        key, X = B.randn(key, B.dtype(L), degree, degree)
+        M = (X + B.transpose(X)) / B.sqrt(2)
+
+        eigv = eigvalsh(M)  # [D]
+
+        if nu == np.inf:
+            proposal = eigv / L
+        else:
+            eigv = eigv / B.sqrt(2 * nu / L**2 + B.sum(rho**2))
+
+            key, chi2_sample = B.randgamma(key, B.dtype(L), 1, alpha=nu, scale=2)
```

Comment:
Repeat the comment `# Gamma(nu/2, 2) distribution is the same as chi2(nu) distribution`